### PR TITLE
osd: avoid unnecessary map lookups in PrimaryLogPG::do_op.

### PR DESCRIPTION
**Pull request by rzarzynski**

### CODE REVIEW

File: src/osd/PrimaryLogPG.cc [hunk 1]
```diff
@@ -2114,26 +2114,27 @@ void PrimaryLogPG::do_op(OpRequestRef& op)
...
-    wait_for_blocked_object(
-      blocked_snap_promote_iter->second->obs.oi.soid,
-      op);
-    return;
-  }
-  if (write_ordered && objects_blocked_on_cache_full.count(head)) {
-    block_write_on_full_cache(head, op);
-    return;
+  if (write_ordered) {
+    map<hobject_t, snapid_t>::iterator blocked_iter =
```
Comment (tchaikov) [inline src/osd/PrimaryLogPG.cc]: nit, could take this chance to use `auto` here. and for better readability, could put

```c++
if (auto blocked_iter = objects_blocked_on_degraded_snap.find(head);
  blocked_iter != objects_blocked_on_degraded_snap.end()) {
  // ...
}
```

so the lexical scope of `blocked_iter` is limited in the `if` clause.

Comment (rzarzynski) [reply src/osd/PrimaryLogPG.cc]: Sure, I will push the refactoring as a separated commit to this PR.

Comment (xiexingguo) [reply src/osd/PrimaryLogPG.cc]: > Sure, I will push the refactoring as a separated commit to this PR.

@rzarzynski 
Would you like to do it here or in a follow-up pr?

Comment (rzarzynski) [reply src/osd/PrimaryLogPG.cc]: @xiexingguo: just pushed the commit.

**Pull request merged**